File: benchmarks/py/bench_match.py
total = 0
for i in range(1000000):
    match i % 10:
        case 0: total += 1
        case 1: total += 2
        case 2: total += 3
        case 3: total += 4
        case 4: total += 5
        case 5: total += 6
        case 6: total += 7
        case 7: total += 8
        case 8: total += 9
        case _: total += 10
print(total)

File: benchmarks/cpp/bench_match.cpp
#include <cstdio>
int main() { long long total=0; for (int i=0; i<1000000; i++) { switch(i%10) { case 0: total+=1; break; case 1: total+=2; break; case 2: total+=3; break; case 3: total+=4; break; case 4: total+=5; break; case 5: total+=6; break; case 6: total+=7; break; case 7: total+=8; break; case 8: total+=9; break; default: total+=10; } } printf("%lld\n", total); return 0; }
